script.recap/resources/lib/core.py

```diff
@@ -0,0 +1,230 @@
+# -*- coding: utf-8 -*-
+import xbmc
+import xbmcaddon
+import xbmcgui
+
+# Import the common settings
+from settings import log
+from settings import Settings
+from recap import Recap
+from viewer import Viewer
+
+ADDON = xbmcaddon.Addon(id='script.recap')
+
+
+#########################
+# Main
+#########################
+def runRecap(previous=False):
+    log("runRecap: Started")
+
+    # First check to see if we have a TV Show already selected
+    showName = None
+    if xbmc.getInfoLabel("ListItem.dbtype") in ['tvshow', 'season', 'episode']:
```

Comment:
Hi @phil65 - I'm afraid I don't follow you for the first point - this should only be called from the tvshow/season/episode windon (not the home window) - do you have an example by any chance? (The context menu should not be shown anywhere else due to the settings in addon.xml)

The the second point - interesting, never really thought about using a set for that sort of lookup - will keep it in mind for future changes.

Thanks, Rob